pygfx/utils/transform.py

```diff
@@ -714,80 +721,76 @@ class RecursiveTransform(AffineBase):
+                child.flag_update()
+
+    @cached
+    def __parent_reference_up(self) -> np.ndarray:
+        """The direction of the reference_up vector expressed in the parent frame."""
+        if self._parent:
+            new_ref = la.vec_transform(self._reference_up, self._parent.inverse_matrix)
+            origin = la.vec_transform((0, 0, 0), self._parent.inverse_matrix)
+            vec = la.vec_normalize(new_ref - origin)
+            vec.flags.writeable = False
+            return vec
+        return self._reference_up_view
 
     @property
-    def last_modified(self) -> int:
-        return max(
-            self._last_modified, self.own.last_modified, self._parent.last_modified
-        )
+    def _parent_reference_up(self) -> np.ndarray:
+        return self.__parent_reference_up
 
-    @cached
-    def __own_reference_up(self) -> np.ndarray:
-        new_ref = la.vec_transform(self._reference_up, self._parent.inverse_matrix)
-        origin = la.vec_transform((0, 0, 0), self._parent.inverse_matrix)
-        vec = la.vec_normalize(new_ref - origin)
-        vec.flags.writeable = False
-        return vec
-
-    @property
-    def _own_reference_up(self) -> np.ndarray:
-        return self.__own_reference_up
-
-    @_own_reference_up.setter
-    def _own_reference_up(self, value):
-        new_ref = la.vec_transform(value, self.parent.matrix)
-        origin = self.parent.position
-        self._reference_up[:] = la.vec_normalize(new_ref - origin)
+    @_parent_reference_up.setter
+    def _parent_reference_up(self, value):
+        if self._parent:
+            new_ref = la.vec_transform(value, self._parent.matrix)
+            origin = self._parent.position
+            self._reference_up[:] = la.vec_normalize(new_ref - origin)
+        else:
+            self._reference_up[:] = la.vec_normalize(value)
         self.flag_update()
-        # Note: since __own_reference_up is only used in a setter in AffineBase
+        # Note: since _parent_reference_up is only used in a setter in AffineBase
         # we do not need to call flag_update() on self.own; all its state and cache
         # remains intact
 
     @property
-    def parent(self) -> AffineBase:
+    def parent(self) -> RecursiveTransform:
         """The transform that preceeds the own/local transform."""
         return self._parent
 
     @parent.setter
-    def parent(self, value):
-        if value is None:
-            self._parent = AffineTransform()
-        else:
-            self._parent = value
+    def parent(self, value: RecursiveTransform):
+        self._parent = value
         self.flag_update()
 
     @cached
-    def _matrix(self):
-        mat = self._parent.matrix @ self.own.matrix
-        mat.flags.writeable = False
-        return mat
+    def _matrix(self) -> np.ndarray:
+        if self._parent:
+            mat = self._parent.matrix @ self.own.matrix
+            mat.flags.writeable = False
+            return mat
+        return self.own.matrix
```

Comment:
I could move this to the property getter actually and skip the cache descriptor overhead